### PR TITLE
Grammar for 5 languages

### PR DESCRIPTION
**Pull request by pchaigno**
PogoScript: [featurist/PogoScript.tmbundle](https://github.com/featurist/PogoScript.tmbundle) - [example in Lightshow](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Ffeaturist%2FPogoScript.tmbundle%2Fmaster%2FSyntaxes%2FPogoScript.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fpchaigno%2Flinguist%2Fmaster%2Fsamples%2FPogoScript%2Fsquashy.pogo&code=)
Opal: [example in Lightshow](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fartifactz%2Fsublime-opal%2Fmaster%2FOpal.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fpchaigno%2Flinguist%2Fmaster%2Fsamples%2FOpal%2FDeepakChopra.opal&code=) - [alternative 1](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fbsingr%2Fopal-tmbundle%2Fmaster%2FSyntaxes%2FOpal.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fpchaigno%2Flinguist%2Fmaster%2Fsamples%2FOpal%2FDeepakChopra.opal&code=) - [alternative 2](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fopal%2Fopal.tmbundle%2Fmaster%2FSyntaxes%2FOpal.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fpchaigno%2Flinguist%2Fmaster%2Fsamples%2FOpal%2FDeepakChopra.opal&code=) - The first one seems to be the only one correctly highlighting comments

The three following grammars have unrecognized licenses:
GAP: [dhowden/gap-tmbundle](https://github.com/dhowden/gap-tmbundle) - [example in Lightshow](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fdhowden%2Fgap-tmbundle%2Fmaster%2FSyntaxes%2FGAP.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fgithub%2Flinguist%2Fmaster%2Fsamples%2FGAP%2FMagic.gi&code=) - License in the README but unrecognized
Papyrus: [Kapiainen/SublimePapyrus](https://github.com/Kapiainen/SublimePapyrus) - [example in Lightshow](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2FKapiainen%2FSublimePapyrus%2Fmaster%2FCore%2FPapyrus.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fgithub%2Flinguist%2Fmaster%2Fsamples%2FPapyrus%2FvSCM_MetaQuestScript.psc&code=) - License not found because it's in the `Core` folder
Propeller Spin: [bitbased/sublime-spintools](https://github.com/bitbased/sublime-spintools) - [example in Lightshow](https://lightshow.githubapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fbitbased%2Fsublime-spintools%2Fmaster%2FSpin.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fraw.githubusercontent.com%2Fgithub%2Flinguist%2Fmaster%2Fsamples%2FPropeller%2520Spin%2FTV.spin&code=) - zlib license in the README but unrecognized

/cc @aroben

### CODE REVIEW

**Comment by pchaigno** [comment]
For the Papyrus license that's not found because it's in a subfolder I was thinking that we could look into all subfolders for the license. But the risk is that would detect license of "subprojects" instead of the right license :/
So maybe for Papyrus the best solution would be to add it to [this list](https://github.com/github/linguist/blob/master/test/test_grammars.rb#L8-L12)?

**Comment by aroben** [comment]
We should teach the test to recognize the zlib license.

I'm checking into whether the GAP license is sufficient.

> For the Papyrus license that's not found because it's in a subfolder I was thinking that we could look into all subfolders for the license. But the risk is that would detect license of "subprojects" instead of the right license :/
> So maybe for Papyrus the best solution would be to add it to this list?

Adding it to the list seems OK to me. Let's rename it to `LICENSE_WHITELIST` and move the comments inside the list. Something like:

```ruby
  LICENSE_WHITELIST = %w[
    # This grammar's MIT license is inside a subdirectory.
    vendor/grammars/SublimePapyrus

    # These grammars have no license but have been grandfathered in. New grammars
    # must have a license that allows redistribution.
    vendor/grammars/Sublime-Lasso
    vendor/grammars/Sublime-REBOL
    vendor/grammars/x86-assembly-textmate-bundle
  ].freeze
```

**Comment by aroben** [comment]
> I'm checking into whether the GAP license is sufficient.

The GAP license is fine. Maybe it should also be added to the whitelist. I don't expect other repos to use that exact license so it probably isn't worth trying to "recognize" it.

**Comment by pchaigno** [comment]
@aroben I added the Papyrus and GAP grammars to the whitelist as you suggested.

To recognize the zlib license I was thinking that we could use the presence of `zlib license` **and** some other part of the license. For that other part do you think `2. Altered source versions must be plainly marked as such` would be sufficient?

File: test/test_grammars.rb [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
-  # These grammars have no license but have been grandfathered in. New grammars
-  # must have a license that allows redistribution.
-  UNLICENSED_GRAMMARS_WHITELIST = %w[
-    vendor/grammars/Sublime-Lasso
-    vendor/grammars/Sublime-REBOL
-    vendor/grammars/x86-assembly-textmate-bundle
+  LICENSE_WHITELIST = [
+    # This grammar's MIT license is inside a subdirectory.
+    "vendor/grammars/SublimePapyrus",
+    "vendor/grammars/gap-tmbundle",
```
Comment (aroben) [inline test/test_grammars.rb]: Could you move `gap-tmbundle` into its own paragraph so it's clearer the subdirectory comment doesn't apply to it? Maybe a comment like "This grammar has a nonstandard but acceptable license" would be helpful too.

**Comment by aroben** [comment]
> To recognize the zlib license I was thinking that we could use the presence of zlib license and some other part of the license.

That sounds good to me.

File: test/test_grammars.rb (hunk 1 above)
Comment (pchaigno) [inline test/test_grammars.rb]: Done :-)

**Comment by pchaigno** [comment]
I added detection of the zlib license and update the pull request.
Thanks for the review @aroben :zap:

**Comment by aroben** [comment]
This looks great. Thanks @pchaigno!

**Comment by aroben** [comment]
This is now live on github.com.

**Pull request merged**